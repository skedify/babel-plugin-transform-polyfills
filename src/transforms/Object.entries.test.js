import compile from '../../test/utils/compile'

describe('Object.entries', () => {
  it('should replace Object.entries with a polyfill', () => {
    const result = compile(
      `
      const c = Object.entries({ a: 1, b: 2 })
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Object.entries({ a: 1, b: 2 })
      const f = Object.entries({ a: 1, b: 2 })
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Object.entries({ a: 1, b: 2 }))
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when using apply', () => {
    const result = compile(
      `
      const c = Object.entries.apply(null, [{ a: 1, b: 2 }])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when using call', () => {
    const result = compile(
      `
      const c = Object.entries.call(null, { a: 1, b: 2 })
    `
    )

    expect(result).toMatchSnapshot()
  })
})
