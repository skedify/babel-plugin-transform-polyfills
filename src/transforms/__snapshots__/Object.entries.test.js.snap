// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Object.entries should not import polyfill more than once 1`] = `
"import _Object_entries from \\"babel-plugin-transform-polyfills/polyfills/Object/entries\\";

const e = _Object_entries({
  a: 1,
  b: 2
});

const f = _Object_entries({
  a: 1,
  b: 2
});"
`;

exports[`Object.entries should replace Object.entries with a polyfill 1`] = `
"import _Object_entries from \\"babel-plugin-transform-polyfills/polyfills/Object/entries\\";

const c = _Object_entries({
  a: 1,
  b: 2
});"
`;

exports[`Object.entries should work when part of a call chain 1`] = `
"import _Object_entries from \\"babel-plugin-transform-polyfills/polyfills/Object/entries\\";
const c = Object.keys(_Object_entries({
  a: 1,
  b: 2
}));"
`;

exports[`Object.entries should work when using apply 1`] = `
"import _Object_entries from \\"babel-plugin-transform-polyfills/polyfills/Object/entries\\";

const c = _Object_entries.apply(null, [{
  a: 1,
  b: 2
}]);"
`;

exports[`Object.entries should work when using call 1`] = `
"import _Object_entries from \\"babel-plugin-transform-polyfills/polyfills/Object/entries\\";

const c = _Object_entries.call(null, {
  a: 1,
  b: 2
});"
`;
